base/linalg/generic.jl

```diff
@@ -299,10 +544,26 @@ end
 vecdot(x::Number, y::Number) = conj(x) * y
 
 dot(x::Number, y::Number) = vecdot(x, y)
+
+"""
+    dot(x, y)
+    ⋅(x,y)
+
+Compute the dot product. For complex vectors, the first vector is conjugated.
+"""
 dot(x::AbstractVector, y::AbstractVector) = vecdot(x, y)
 
 ###########################################################################################
 
+"""
+    rank(M[, tol::Real])
+
+Compute the rank of a matrix by summing the singular
```

Comment:
`rank` docstring - it's not really summing the singular values themselves, it's counting the number of them that have magnitude greater than `tol`